crates/amaru-ledger/src/store/columns/pools.rs

```diff
@@ -102,14 +105,17 @@ impl Row {
                         .unwrap_or_else(|| unreachable!("pre-condition: needs_update"))
                         .future_params
                         .last();
+
                     assert_eq!(
                         last,
                         Some(&(None, current_epoch)),
                         "invariant violation: most recent retirement is not last certificate: {:?}",
                         last,
                     );
+
                     *pool = None;
-                    return;
+
+                    return Some(refund);
```

Comment:
We now return the credential that needs refund when the pool is retired. This credential may or may not exist anymore. The case where it no longer exists is handled by the refund function.